virtcontainers/clh_test.go

```diff
@@ -501,6 +502,53 @@ func TestClhCliVsock(t *testing.T) {
 
 }
 
+func TestCheckVersion(t *testing.T) {
+	clhConfig := newClhConfig()
+	clh := &cloudHypervisor{
+		config: clhConfig,
+	}
+	assert := assert.New(t)
+	testcases := []struct {
+		name  string
+		major int
+		minor int
+		pass  bool
+	}{
+		{
+			name:  "minor lower than supported version",
+			major: supportedMajorVersion,
+			minor: 2,
+			pass:  true,
+		},
+		{
+			name:  "minor equal to supported version",
+			major: supportedMajorVersion,
+			minor: supportedMinorVersion,
+			pass:  true,
+		},
+		{
+			name:  "major exceeding supported version",
+			major: 1,
+			minor: supportedMinorVersion,
+			pass:  false,
+		},
+	}
+	for _, tc := range testcases {
+		clh.version = CloudHypervisorVersion{
+			Major:    tc.major,
+			Minor:    tc.minor,
+			Revision: 0,
+		}
+		err := clh.checkVersion()
+		msg := fmt.Sprintf("%s, result: %v", tc.name, err)
```

Comment:
Please log `tc` as shown in the document. I would also add `chl.version` to `msg` personally. Think about how easy it is to debug a failing CI test running on a remote system where you only have access to the test log...

Reply:
I have updated the test.